src/keboola/http_client/http.py

```diff
@@ -0,0 +1,664 @@
+    AUTH_HEADER = {
+        'x-storageapi-token': '1234-STORAGETOKENSTRING'
+    }
+    DEFAULT_PARAMS = {
+        'include': 'columns'
+    }
+    DEFAULT_HEADER = {
+        'Content-Type': 'application/json'
+    }
+
+    cl = HttpClient(BASE_URL, default_http_header=DEFAULT_HEADER,
+                    auth_header=AUTH_HEADER, default_params=DEFAULT_PARAMS)
+
+    files_response = cl.get("files", params={"showExpired": True})
+    ```
+
+    """
+
+    def __init__(self, base_url: str, max_retries: int = 10, backoff_factor: float = 0.3,
+                 status_forcelist: Tuple[int, ...] = (500, 502, 504), default_http_header: Dict = None,
+                 auth_header: Dict = None, auth: Tuple = None, default_params: Dict = None,
+                 method_whitelist: Tuple = METHOD_RETRY_WHITELIST):
+        """
+        Create an endpoint.
+
+        Args:
+            base_url: The base URL for this endpoint. e.g. https://exampleservice.com/api_v1/
+            max_retries: Total number of retries to allow.
+            backoff_factor:  A back-off factor to apply between attempts.
+            status_forcelist:  A set of HTTP status codes that we should force a retry on. e.g. [500,502]
+            default_http_header: Default header to be sent with each request
+                eg. {
+                        'Content-Type' : 'application/json',
+                        'Accept' : 'application/json'
+                    }
+            auth_header: Auth header to be sent with each request
+                eg. {'Authorization': 'Bearer ' + token}
+            auth: Default Authentication tuple or object to attach to (from  requests.Session().auth).
+                eg. auth = (user, password)
+            default_params: default parameters to be sent with each request eg. {'param':'value'}
+            method_whitelist (tuple): Set of upper-cased HTTP method verbs that we should retry on.
+        """
+        if base_url is None:
+            raise ValueError("Base URL is required.")
+        # Add trailing slash because of nature of urllib.parse.urljoin()
+        self.base_url = base_url if base_url.endswith('/') else base_url + '/'
+        self.max_retries = max_retries
+        self.backoff_factor = backoff_factor
+        self.status_forcelist = status_forcelist
+        self._auth = auth
+        self._auth_header = auth_header if auth_header else {}
+        self._default_header = default_http_header if default_http_header else {}
+        self._default_params = default_params
+        self.method_whitelist = method_whitelist
+
+    def requests_retry_session(self, session=None):
+        session = session or requests.Session()
+        retry = Retry(
+            total=self.max_retries,
+            read=self.max_retries,
+            connect=self.max_retries,
+            backoff_factor=self.backoff_factor,
+            status_forcelist=self.status_forcelist,
+            method_whitelist=self.method_whitelist
```

Comment:
Btw @davidesner , this was changed to `alllowed_methods` on purpose, because there's a [deprecation warning](https://github.com/MarketSquare/robotframework-requests/issues/307) for `method_whitelist` parameter. So I suppose, we should change this back to `allowed_methods` what do you think?

Reply:
Both `allowed_methods` and `method_whitelist` have the same usage, [see repo](https://github.com/urllib3/urllib3/blob/1.26.2/src/urllib3/util/retry.py#L257).

Reply:
fixed in commit 5f4e6b